File: online_store/domain/models/ShoppingCart.py
from online_store.domain.models.products.ProductInterface import ProductInterface
from online_store.domain.models.Catalog import Catalog

class ShoppingCart:
    def __init__(self):
        self.items = []

    def add_item(self, item:ProductInterface):
        self.items.append(item)

    def remove_item(self, item:ProductInterface):
        self.items.remove(item)

    def calculateTotalPrice(self) -> float:
        if not self.items:
            return None
        total_price = sum(item.getPrice() for item in self.items)
        return total_price

File: tmpsjava/online_shop/src/domain/models/ShoppingCart.java
package domain.models;

import domain.models.products.ProductInterface;

import java.util.ArrayList;
import java.util.List;

public class ShoppingCart implements ShoppingCartInterface {
    private List<ProductInterface> items;

    public ShoppingCart() {
        this.items = new ArrayList<>();
    }

    public void add_item(ProductInterface item) {
        this.items.add(item);
    }

    public void remove_item(ProductInterface item) {
        this.items.remove(item);
    }
    @Override
    public Float calculateTotalPrice() {
        if (this.items.isEmpty()) {
            return 0.0f;
        }

        float total_price = 0;
        for (ProductInterface item : this.items) {
            total_price += item.getPrice();
        }
        return total_price;
    }

    @Override
    public List<ProductInterface> getItems() {
        return this.items;
    }

    public void clear(){
        this.items = new ArrayList<>();
    }

}
